examples/catalog/bin/save_screenshots.dart

```diff
@@ -0,0 +1,139 @@
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+import 'dart:math' as math;
+
+import 'package:image/image.dart';
+
+String authorizationToken;
+
+class UploadError extends Error {
+  UploadError(this.message);
+  final String message;
+  @override
+  String toString() => message;
+}
+
+void logMessage(String s) { print(s); }
+
+class Upload {
+  Upload(this.fromPath, this.largeName, this.smallName);
+
+  static math.Random random;
+  static final String uriAuthority = 'www.googleapis.com';
+  static final String uriPath = 'upload/storage/v1/b/flutter-catalog/o';
+
+  final String fromPath;
+  final String largeName;
+  final String smallName;
+
+  List<int> largeImage;
+  List<int> smallImage;
+  bool largeImageSaved;
+  int retryCount = 0;
+  bool isComplete = false;
+
+  Duration get timeLimit {
```

Comment:
Could you add a comment explaining that it does (what I believe is) exponential back-off?